### PR TITLE
Add HasDisciplineLearned() to Perl/Lua.

### PR DESCRIPTION
**Pull request by KinglyKrab**
- Add $client->HasDisciplineLearned(spell_id) to Perl.
- Add client:HasDisciplineLearned(spell_id) to Lua.

### CODE REVIEW

File: zone/effects.cpp [hunk 1]
```diff
@@ -678,6 +678,16 @@ void Client::ResetDisciplineTimer(uint32 timer_id) {
 	SendDisciplineTimer(timer_id, 0);
 }
 
+bool Client::HasDisciplineLearned(uint16 spell_id) {
+	bool has_learned = false;
+	for (auto index = 0; index < MAX_PP_DISCIPLINES; ++index) {
+		if (GetPP().disciplines.values[index] == spell_id) {
+			has_learned = true;
```
Comment (mackal) [inline zone/effects.cpp:685]: We can just return true here and return false later.

**Review by mackal — CHANGES_REQUESTED** [review]

File: zone/effects.cpp (hunk 1 above)
Comment (KinglyKrab) [reply zone/effects.cpp:685]: Figured one point of return was better than 2.

**Review by Akkadius — APPROVED** [review]

File: zone/effects.cpp (hunk 1 above)
Comment (mackal) [reply zone/effects.cpp:685]: well, if the first discipline checked is the one you're searching for, it's gonna still search the rest of them. Ideally you return early, you stop searching. Currently this always has the same complexity, where on average we should be hitting O(n/2).

**Comment by Akkadius** [comment]
Current pr has conflicts to resolve, also has suggestions. Lgtm otherwise

File: zone/effects.cpp (hunk 1 above)
Comment (mackal) [reply zone/effects.cpp:685]: We could also just use std::find_if.

Comment (KinglyKrab) [reply zone/effects.cpp:685]: I'll just add a break in the loop.

**Review by mackal — APPROVED** [review]

**Comment by mackal** [comment]
Conflicts still nee to be resolved it appears.

**Comment by KinglyKrab** [comment]
Merging based on approval and passing all checks. Going to squash to one commit.

**Pull request merged**